builtin/providers/vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -127,6 +129,66 @@ func TestAccVSphereVirtualMachine_dhcp(t *testing.T) {
 	})
 }
 
+func TestAccVSphereVirtualMachine_createInFolder(t *testing.T) {
+	var vm virtualMachine
+	var locationOpt string
+	var datastoreOpt string
+	var datacenter string
+
+	if v := os.Getenv("VSPHERE_DATACENTER"); v != "" {
+		locationOpt += fmt.Sprintf("    datacenter = \"%s\"\n", v)
+		datacenter = v
+	}
+	if v := os.Getenv("VSPHERE_CLUSTER"); v != "" {
+		locationOpt += fmt.Sprintf("    cluster = \"%s\"\n", v)
+	}
+	if v := os.Getenv("VSPHERE_RESOURCE_POOL"); v != "" {
+		locationOpt += fmt.Sprintf("    resource_pool = \"%s\"\n", v)
+	}
+	if v := os.Getenv("VSPHERE_DATASTORE"); v != "" {
+		datastoreOpt = fmt.Sprintf("        datastore = \"%s\"\n", v)
+	}
+	template := os.Getenv("VSPHERE_TEMPLATE")
+	label := os.Getenv("VSPHERE_NETWORK_LABEL_DHCP")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVSphereVirtualMachineDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				PreConfig: func() { assureVSphereFolderExists(datacenter, "terraform-test-folder") },
```

Comment:
Looks like this test might leak the "terraform-test-folder" folder into whatever environment it is run in?

You could clean it up by switching the `CheckDestroy` above to a `resource.ComposeTestCheckFunc` with the existing check along with something like `assureVSphereFolderRemoved`.